audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -286,32 +281,31 @@ public int onStartCommand(final Intent intent, int flags, int startId) {
     }
 
     public void stop() {
+        notificationCreated = false;
         deactivateMediaSession();
-        stopSelf();
+        stopForeground(true);
     }
 
     @Override
     public void onDestroy() {
         super.onDestroy();
         listener.onDestroy();
         listener = null;
-        mediaMetadata = null;
-        artBitmap = null;
-        queue.clear();
         mediaMetadataCache.clear();
-        actions.clear();
         artBitmapCache.evictAll();
-        compactActionIndices = null;
+        queue.clear();
+        mediaMetadata = null;
+        artBitmap = null;
+        notificationCreated = false;
+        playbackState = null;
+        contentIntent = null;
+        notificationChannelId = null;
+        config = null;
         releaseMediaSession();
-        stopForeground(!config.androidResumeOnClick);
```

Comment:
It might be related to https://github.com/ryanheise/audio_service/issues/701. It seems that this configuration parameter shouldn't affect whether the notification is removed on destroy. I strongly feel that when service is stopped whether the notification is removed should be controlled by some standalone config parameter

Reply:
I think `androidResumeOnClick` was a poorly named option in retrospect, but in all practicality it serves exactly the same purpose as trying to keep the notification around. Why else would we keep it around if not to be able to resume the session?

Reply:
I also don't understand why you're deleting this.

Reply:
In any case, since this is not related to the other changes, it's best done as a separate PR which I can choose whether to merge independently of the other changes.

Reply:
> I think androidResumeOnClick was a poorly named option in retrospect, but in all practicality it serves exactly the same purpose as trying to keep the notification around. Why else would we keep it around if not to be able to resume the session?

It has no mention in documentation regarding this behavior. Also, it doesn't seem to be working then, because when set to true, it doesn't behave as you described. I'm not sure what the issue right now then, already filed https://github.com/ryanheise/audio_service/issues/638 or missing documentation.

(I removed it because I moved it below and set to `true`, since practially it seems to have no difference as described above)